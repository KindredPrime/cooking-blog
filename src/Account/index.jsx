import { Component } from 'react';
import PropTypes from 'prop-types';
import PostsList from '../PostsList/index';
import UserCommentsList from '../UserCommentsList/index';
import * as API from '../apiCalls';

class Account extends Component {
  state = {
    blogPosts: null,
    comments: null
  };

  componentDidMount() {
    const { username } = this.props;

    API.getCommentsByUser(username)
      .then((comments) => {
        this.setState({
          comments
        });
      })
      .catch(console.log);

    API.getBlogPostsByUser(username)
      .then((blogPosts) => this.setState({ blogPosts }))
      .catch(console.log);
  }

  componentWillUnmount() {
    API.abortTasks();
  }

  render() {
    const { username, email } = this.props;
    const { blogPosts, comments } = this.state;

    return (
      <main className="Account">
        <h1>{username}</h1>

        <p>Email: {email}</p>

        {blogPosts && <PostsList initialBlogPosts={blogPosts} />}

        {comments && <UserCommentsList initialComments={comments} />}
      </main>
    );
  }
}

Account.propTypes = {
  username: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired
};

export default Account;