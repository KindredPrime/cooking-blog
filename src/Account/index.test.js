import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Account from './index';

describe('Account Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Account username="username" email="email" />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    render(
      <BrowserRouter>
        <Account username="username" email="email" />
      </BrowserRouter>
    );

    expect(document.body).toMatchSnapshot();
  });
});