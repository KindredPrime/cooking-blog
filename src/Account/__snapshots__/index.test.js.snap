// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Component renders the UI as expected 1`] = `
<body>
  <div>
    <main
      class="Account"
    >
      <h1>
        username
      </h1>
      <p>
        Email: 
        email
      </p>
    </main>
  </div>
</body>
`;
